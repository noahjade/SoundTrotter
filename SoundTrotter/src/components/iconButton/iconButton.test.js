import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";

import IconButton from ".";

test("Correctly renders next button", () => {
  const view = renderer.create(<IconButton type="next" />);
  expect(view).toMatchSnapshot();
});

test("Correctly renders prev button", () => {
  const view = renderer.create(<IconButton type="previous" />);
  expect(view).toMatchSnapshot();
});

test("Correctly renders play button", () => {
  const view = renderer.create(<IconButton type="play" />);
  expect(view).toMatchSnapshot();
});

test("Correctly renders pause button", () => {
  const view = renderer.create(<IconButton type="pause" />);
  expect(view).toMatchSnapshot();
});

test("Correctly renders disabled button", () => {
  const view = renderer.create(<IconButton isDisabled={true} />);
  expect(view).toMatchSnapshot();
});

test("Click handler is called when button is clicked", () => {
  const mockClickHandler = jest.fn();
  const { getByText } = render(
    <IconButton type="next" clickHandler={mockClickHandler} />
  );
  const svg = getByText("next.svg");
  fireEvent.click(svg);
  expect(mockClickHandler).toHaveBeenCalledTimes(1);
});
