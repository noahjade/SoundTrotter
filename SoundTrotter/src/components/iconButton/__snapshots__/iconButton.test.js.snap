// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly renders disabled button 1`] = `
<button
  className="btn"
  disabled={true}
  id="play-btn"
>
  <svg>
    play.svg
  </svg>
</button>
`;

exports[`Correctly renders next button 1`] = `
<button
  className="btn"
  id="next-btn"
>
  <svg>
    next.svg
  </svg>
</button>
`;

exports[`Correctly renders pause button 1`] = `
<button
  className="btn"
  id="pause-btn"
>
  <svg>
    pause.svg
  </svg>
</button>
`;

exports[`Correctly renders play button 1`] = `
<button
  className="btn"
  id="play-btn"
>
  <svg>
    play.svg
  </svg>
</button>
`;

exports[`Correctly renders prev button 1`] = `
<button
  className="btn"
  id="previous-btn"
>
  <svg>
    prev.svg
  </svg>
</button>
`;
